nautobot/extras/filters.py

```diff
@@ -571,6 +573,22 @@ class Meta:
         fields = ("id", "name", "slug", "description")
 
 
+class DynamicGroupMembershipFilterSet(NautobotFilterSet):
+    q = SearchFilter(
+        filter_predicates={
+            "operator": "icontains",
+            "group__name": "icontains",
+            "group__slug": "icontains",
+            "parent_group__name": "icontains",
+            "parent_group__slug": "icontains",
+        },
+    )
+
+    class Meta:
+        model = DynamicGroupMembership
+        fields = ("id", "group", "parent_group", "operator", "weight")
```

Comment:
Should we use `NaturalKeyOrPKMultipleChoiceFilter` for `group` and `parent_group`?

Reply:
Yes, in fact. Thank you.